core/src/test/java/edu/wpi/grip/core/http/GripServerTest.java

```diff
@@ -0,0 +1,166 @@
+import edu.wpi.grip.core.MockPipeline.MockProjectSettings;
+
+import edu.wpi.grip.core.http.GripServer.HttpServerFactory;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.Arrays;
+import java.util.function.Supplier;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpVersion;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.BasicHttpEntity;
+import org.apache.http.impl.client.DefaultHttpClient;
+import org.apache.http.params.CoreProtocolPNames;
+import org.apache.http.util.EntityUtils;
+import org.junit.After;
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+/**
+ *
+ */
+public class GripServerTest {
+
+    private static final int GRIP_SERVER_TEST_PORT = 8080;
+    private final DefaultHttpClient client;
+
+    private HttpServer server;
+    private int port;
+
+    public static class TestServerFactory implements HttpServerFactory {
+
+        @Override
+        public HttpServer create(int port) {
+            HttpServer server;
+            for (int offset = 0;; offset++) {
+                try {
+                    server = HttpServer.create(new InetSocketAddress("localhost", port + offset), 1);
+                    break;
+                } catch (IOException e) {
+                    // That port is taken -- keep trying different ports
+                }
+            }
+            return server;
+        }
+
+    }
+
+    // DON'T USE INJECTION FOR THIS
+    private GripServer instance;
+
+    public GripServerTest() {
+        for (int portOffset = 0;; portOffset++) {
+            try {
+                server = HttpServer.create(new InetSocketAddress("localhost", GRIP_SERVER_TEST_PORT + portOffset), 1);
+                port = GRIP_SERVER_TEST_PORT + portOffset;
+                break;
+            } catch (IOException ex) {
+                // That port is taken -- keep trying different ports
+            }
+        }
+        MockProjectSettings mockSettings = new MockProjectSettings();
+        mockSettings.setServerPort(port);
+        instance = new GripServer((ignore) -> server, new MockPipeline(mockSettings));
+        instance.start();
+
+        client = new DefaultHttpClient();
+        client.getParams().setParameter(CoreProtocolPNames.PROTOCOL_VERSION, HttpVersion.HTTP_1_1);
+    }
+
+    /**
+     * Test of GetHandler methods, of class GripServer.
+     */
+    @Test
+    public void testGetHandlers() {
+        String path = "/testGetHandlers";
+        GetHandler handler = params -> path;
+        instance.addGetHandler(path, handler);
+        instance.addDataSupplier(path, () -> path);
+        try {
+            String data = doGet(path);
+            assertEquals(data, path);
+        } catch (IOException ex) {
+            fail(ex.getMessage());
+        } finally {
+            instance.removeGetHandler(path); // cleanup
```

Comment:
You shouldn't need to do this? The test constructor gets called again each time for each test.